### PR TITLE
fix: Update diesel dependency

### LINKED ISSUE #70: diesel-adapter is dependent on an old version of diesel, which depends on vulnerable versions of libsqlite3-sys
`cargo deny` now flags `diesel-adapter` as contributing a vulnerable `libsqlite3-sys`.

```
= ID: RUSTSEC-2022-0090
    = Advisory: https://rustsec.org/advisories/RUSTSEC-2022-0090
    = It was sometimes possible for SQLite versions >= 1.0.12, < 3.39.2 to allow an array-bounds overflow when large string were input into SQLite's `printf` function.
      
      As `libsqlite3-sys` bundles SQLite, it is susceptible to the vulnerability. `libsqlite3-sys` was updated to bundle the patched version of SQLite [here](https://github.com/rusqlite/rusqlite/releases/tag/sys0.25.1).
    = Announcement: https://nvd.nist.gov/vuln/detail/CVE-2022-35737
    = Solution: Upgrade to >=0.25.1
    = libsqlite3-sys v0.22.2
      ├── diesel v1.4.8
      │   └── diesel-adapter v0.9.0
      │       └── watchful-hub v3.0.0
```

Is `diesel-adapter` maintained? The latest version of `diesel-adapter` is dependent on a release of `diesel` that was released a while ago.

### PR DESCRIPTION
**Pull request by peschkaj**
This updates Diesel and transitive dependencies to remove a SQLite vulnerability and close #70 

The `eq_empty` macro is removed and I believe these changes maintain the original invariants. Happy to change if they don't.

### CODE REVIEW

**Comment by hsluoyz** [comment]
@peschkaj plz fix CI:

image

**Comment by peschkaj** [comment]
> @peschkaj plz fix CI:
> 
> image

It looks like, from the logs, that the GitHub action isn't able to find a binary version of tarpaulin. As far as I can tell, this is in a downstream dependency.

I'm able to use tarpaulin locally and verify this:

```
Mar 11 11:10:40.545  INFO cargo_tarpaulin::process_handling::linux: Launching test
Mar 11 11:10:40.545  INFO cargo_tarpaulin::process_handling: running /home/jeremiah/src/diesel-adapter/target/debug/deps/diesel_adapter-78a9d3144878db07

running 1 test
test adapter::tests::test_adapter ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.14s

Mar 11 11:10:41.375  INFO cargo_tarpaulin::report: Coverage Results:
|| Uncovered Lines:
|| src/actions.rs: 95, 116, 137, 143, 157, 161-162, 164-165, 171-172, 174-176, 182-183, 185-188, 224, 227, 229, 231-232, 240, 250, 253, 255, 263, 287, 290, 292
|| src/adapter.rs: 30, 32, 36, 39-40, 47, 75, 79, 90, 113, 119, 144, 153, 158, 180-181, 184, 188-190, 200, 202, 207, 210, 236, 266, 278, 287, 290, 314, 327, 347, 354, 372, 379, 399, 408, 419, 423-424
|| src/error.rs: 11-12, 15-17, 22-23, 26-28, 33-35, 39-41
|| Tested/Total Lines:
|| src/actions.rs: 79/112
|| src/adapter.rs: 128/168
|| src/error.rs: 0/16
|| src/macros.rs: 6/6
||
70.53% coverage, 213/302 lines covered

diesel-adapter on  master is 📦 v0.9.0 via 🦀 v1.67.1 took 17s
❯
```

**Comment by hsluoyz** [comment]
@peschkaj we need to make all CI pass, otherwise the PR cannot be further processed.

**Comment by peschkaj** [comment]
@hsluoyz it looks like the CI is failing for reasons unrelated to this PR - all of the CI is timing out except one macOS run that I can't tell why it fails. 

I've kicked off another run and hopefully that works.

**Comment by hsluoyz** [comment]
@peschkaj CI failed again. To prove that the CI error is unrelated with this PR, can you make another PR trying to fix some typo in README, to see if this happens again?

**Comment by peschkaj** [comment]
@hsluoyz with pleasure. I've created https://github.com/casbin-rs/diesel-adapter/pull/73 with a small set of README fixes.

File: src/actions.rs [hunk 1]
```diff
@@ -154,95 +154,91 @@ pub fn remove_filtered_policy(
...
+            .filter(
                 ptype
                     .eq(pt)
-                    .and(eq_empty!(&field_values[0], v4))
-                    .and(eq_empty!(&field_values[1], v5)),
-            ),
-        )
-        .into_boxed()
+                    .and(v4.is_null().or(v4.eq(&field_values[0])))
+                    .and(v4.is_null().or(v5.eq(&field_values[1]))),
```
Comment (ondrowan) [inline src/actions.rs:166]: I think there's a typo here and it should be `v5.is_null()`.

Comment (hsluoyz) [reply src/actions.rs:166]: @peschkaj

**Comment by davidstanley01** [comment]
@ondrowan or @hsluoyz Any progress on this?

**Pull request closed without merging**